cpp/src/arrow/testing/gtest_util.h

```diff
@@ -345,4 +346,71 @@ void AssertSortedEquals(std::vector<T> u, std::vector<T> v) {
   ASSERT_EQ(u, v);
 }
 
+// Mixin used to generate trivial Array/RecordBatch/RecordBatchReader
+class ArrowBaseFixtureMixin {
```

Comment:
I think we should avoid "fixtures" that have no internal state. They complicate usage vs. a bunch of utility functions.